pkg/apis/ceph.rook.io/v1/network.go

```diff
@@ -39,6 +39,9 @@ func (n *NetworkSpec) IsHost() bool {
 }
 
 func ValidateNetworkSpec(clusterNamespace string, spec NetworkSpec) error {
+	if spec.HostNetwork && (spec.Provider != NetworkProviderDefault) {
```

Comment:
do we ever touch this code, as we already have the x-validation for this in the crd?

Reply:
@parth-gr  wrote:
> do we ever touch this code, as we already have the x-validation for this in the crd?

good question. This started with the request to adjust this code and then later the request was added to update the x-validation rule. @BlaineEXE , @subhamkrai, wdyt?

Reply:
These validations are still important for 
1. users who don't have a K8s version that supports VAPs
2. any applications that use Rook's API as a library

Reply:
Just because we are checking input values in the beginning of the system, it doesn't mean that checking input values later is not valuable. I have seen others ask this same question in other scenarios, and I want to keep stressing that the idea that "we checked the input once, so we don't need to do it again" is likely to lead to more errors in **any** software system. Input validation is important at all layers of a software system (not just API layers) and is discussed widely in books, blogs, and threads as a good practice. 

Input validation is almost never a computationally expensive operation, and we have almost no reason to do work to avoid it. A large software system can have many places in the code where the values might be modified accidentally or purposefully, and having validation available means that the layer where those values are most important can still verify their validity before it proceeds to configure the system.

The reason this is important for Rook isn't to help the project **now**. It is to help the project throughout its entire maintenance lifespan. If someone makes code changes that affect these values in a seemingly-unrelated portion of the Rook code, Rook's later code paths can still detect an internal configuration error before applying network changes. This likely means that the seemingly-unrelated change will be flagged in CI as causing an error, meaning it won't propagate to end users. Or, if the change does sneak through CI, it at least means that production users won't be affected by the change that otherwise might cause a bug in their Ceph cluster configuration.

Reply:
Understood, that was a great informatve read.
Thanks